Code/Framework/AzFramework/Platform/Android/AzFramework/Device/DeviceAttributesCommon_Android.cpp

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Android/JNI/JNI.h>
+#include <AzCore/Android/JNI/Object.h>
+#include <AzFramework/Device/DeviceAttributes.h>
+
+namespace AzFramework
+{
+    DeviceAttributeDeviceModel::DeviceAttributeDeviceModel()
+    {
+        static constexpr const char* JavaFieldName = "MODEL";
+        AZ::Android::JNI::Object obj("android/os/Build");
+        obj.RegisterStaticField(JavaFieldName, "Ljava/lang/String;");
+        m_value = obj.GetStaticStringField(JavaFieldName);
+    }
+
+    DeviceAttributeRAM::DeviceAttributeRAM()
+    {
+        static constexpr const char* JavaFuntionNameGetDeviceRamInGB = "GetDeviceRamInGB";
+        AZ::Android::JNI::Object obj("com/amazon/lumberyard/AndroidDeviceManager");
+        obj.RegisterStaticMethod(JavaFuntionNameGetDeviceRamInGB, "()F");
+        m_value = obj.InvokeStaticFloatMethod(JavaFuntionNameGetDeviceRamInGB);

```

Comment:
JNI calls are such a pain.
Especially since Android tends to change API calls every couple of SDKs

Reply:
I pulled this from the LY implementation and it still works.